internal/services/formatters/generic/trivy/formatter.go

```diff
@@ -133,47 +139,38 @@ func (f *Formatter) getDockerConfig(cmd, projectSubPath string) *docker.Analysis
 	return analysisData.SetImage(f.GetCustomImageByLanguage(languages.Generic), images.Generic)
 }
 
-func (f *Formatter) parseOutput(output, cmd, projectSubPath string) error {
+func (f *Formatter) parseOutput(output, projectSubPath string) error {
 	report := new(trivyOutput)
-
 	if output == "" {
 		return nil
 	}
-
 	if err := json.Unmarshal([]byte(output), report); err != nil {
 		return err
 	}
-
 	for _, result := range report.Results {
-		f.addVulnerabilities(cmd, result, filepath.Join(projectSubPath, result.Target))
+		f.addVulnerabilitiesOutput(result.Vulnerabilities, filepath.Join(projectSubPath, result.Target), projectSubPath)
+		if result.Misconfigurations != nil {
+			f.addMisconfigurationOutput(result.Misconfigurations, filepath.Join(projectSubPath, result.Target))
+		}
 	}
-
 	return nil
 }
 
-func (f *Formatter) addVulnerabilities(cmd string, result *trivyOutputResult, path string) {
-	switch cmd {
-	case CmdFs:
-		f.addVulnerabilitiesOutput(result.Vulnerabilities, path)
-	case CmdConfig:
-		f.addVulnerabilitiesOutput(result.Vulnerabilities, path)
-		f.addMisconfigurationOutput(result.Misconfigurations, path)
-	}
-}
-
 // nolint: funlen // needs to be bigger
-func (f *Formatter) addVulnerabilitiesOutput(vulnerabilities []*trivyVulnerability, target string) {
+func (f *Formatter) addVulnerabilitiesOutput(vulnerabilities []*trivyVulnerability, target, projectSubPath string) {
 	for _, vuln := range vulnerabilities {
 		addVuln := f.getVulnBase()
 		addVuln.RuleID = vuln.VulnerabilityID
-		addVuln.Code = fmt.Sprintf("%s v%s", vuln.PkgName, vuln.InstalledVersion)
-		_, _, line, err := file.GetDependencyInfo(addVuln.Code, filepath.Join(f.GetConfigProjectPath(), target))
+		dependencyInfo, err := file.GetDependencyInfo(
+			[]string{vuln.PkgName, vuln.InstalledVersion},
+			[]string{filepath.Join(f.GetConfigProjectPath(), target)},
+		)
 		if err != nil {
-			f.SetAnalysisError(err, tools.Trivy, err.Error(), "")
+			f.SetAnalysisError(err, tools.Trivy, "", projectSubPath)
```

Comment:
Why not use err.Error() here? The error message will be empty.

Reply:
because is it not correct, the parameter that is empty expects the output of the analysis that is not needed here. And the error was already added in the first parameter, adding it again here will only be more confusing

Reply:
Oh, my bad, sorry.